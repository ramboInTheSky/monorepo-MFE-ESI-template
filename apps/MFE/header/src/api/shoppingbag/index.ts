import Logger from "@monorepo/core-logger"
import qs from "qs"
import {ESB} from "@monorepo/eventservice/lib/esb"
import {GeneralCallbackContractModel} from "@monorepo/eventservice"
import {axiosInstance as axios} from "../../utils/axios"
import setMainSiteBagCookie from "../../utils/setMainSiteBagCookie"

export const ShoppingBagApiGet = (
    apiModel: any,
    url: string,
    callbackEventService: ESB,
    sdkCallback: (res) => void,
) => {
    const cacheBustedParams = {...apiModel, _: Date.now()}
    axios
        .get(url, {
            withCredentials: true,
            params: cacheBustedParams,
        })
        .then(response => {
            handleSuccess(response, callbackEventService)
            sdkCallback(response)
            setMainSiteBagCookie(response)
        })
        .catch(error => handleError(error, callbackEventService))
}

export const ShoppingBagApiPost = (apiModel: any, url: string, callbackEventService: ESB, eventId?: string) => {
    const params = qs.stringify(apiModel)

    axios
        .post(url, params, {
            withCredentials: true,
        })
        .then(response => {
            handleSuccess(response, callbackEventService, eventId)
            setMainSiteBagCookie(response)
        })
        .catch(error => handleError(error, callbackEventService, eventId))
}

const handleSuccess = (response, callbackEventService: ESB, eventId?: string) => {
    const callbackEventData: GeneralCallbackContractModel = {
        eventId,
        data: response.data,
        status: response.status,
        success: true,
        textStatus: response.statusText,
    }

    callbackEventService.publish(callbackEventData)
}

const handleError = (error, callbackEventService: ESB, eventId?: string) => {
    let callbackEventData: GeneralCallbackContractModel
    if (error.response) {
        callbackEventData = {
            eventId,
            data: error.response.data,
            status: error.response.status,
            success: false,
            textStatus: error.response.statusText,
        }
    } else if (error.request) {
        callbackEventData = {
            eventId,
            data: null,
            status: error.request.status,
            success: false,
            textStatus: error.request.statusText,
        }
    } else {
        callbackEventData = {
            eventId,
            data: null,
            status: 500,
            success: false,
            textStatus: error.message,
        }
    }
    Logger.error(new Error(error))
    callbackEventService.publish(callbackEventData)
}
