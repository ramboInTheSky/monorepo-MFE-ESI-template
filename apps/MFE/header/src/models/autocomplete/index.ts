export class SuggestionItem {
    q = ""
    dq = ""
}

export class ProductItem {
    sale_price = 0
    url = ""
    pid = ""
    thumb_image = ""
    title = ""
}
export class AutoCompleteParameters {
    accountId = ""
    domainKey = ""
    authKey = ""
}

export class AutoCompleteProducts {
    pid = ""
    sale_price = 0
    thumb_image = ""
    title = ""
    url = ""
}

export class AutoCompleteStateModel {
    q = ""
    suggestions: SuggestionItem[] | null = []
    numFound = 0
    products: AutoCompleteProducts[] | null = []
    parameters: AutoCompleteParameters = new AutoCompleteParameters()
    isLoading = false
}

export class AutoCompleteInitial {
    response = new AutoCompleteStateModel()
}
