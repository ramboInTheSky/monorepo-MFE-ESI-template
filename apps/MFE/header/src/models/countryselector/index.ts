export class CountryLanguage {
    Name = ""
    Value = ""
    TargetUrl = ""
    AccountDomainUrl = ""
}

export class Country {
    Name = ""
    Country = ""
    CountryCode = ""
    DisplayText = ""
    NativeCountryText = ""
    RedirectUrl = ""
    DefaultLanguageName = ""
    Languages: CountryLanguage[] = []
    DisplaySequenceAttribute = ""
    PromotedFlagIndex = 0
    Region = ""
    iconUrl = ""
    DomainType = ""
    HideInDropdown = false
    HideInPage = false
    IsROW = false
}
