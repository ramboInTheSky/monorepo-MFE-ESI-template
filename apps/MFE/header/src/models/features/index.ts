import {FeatureSwitchData} from "./searchBar"

export enum SupportedFeatureTypes {
    SearchBar = "SearchBar",
}

export class SupportedFeatures {
    public features = new SupportedFeatureSwitch()
}

export class SupportedFeatureSwitch {
    public SearchBar = new FeatureSwitchData()
}
