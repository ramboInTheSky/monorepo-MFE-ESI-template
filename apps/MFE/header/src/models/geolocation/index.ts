export class GeolocationModel {
    ISOCountryCode?: string | null
    ISOCountryName?: string | null
    RedirectUrl?: string | null
}
