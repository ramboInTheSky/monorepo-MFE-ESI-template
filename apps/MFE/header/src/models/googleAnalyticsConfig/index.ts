export class GoogleAnalyticsAccount {
    account = ""
    environmentKeys = new EnvironmentKeys()
}

class EnvironmentKeys {
    dev = ""
    uat = ""
    conv = ""
    live = ""
}

export class GoogleAnalyticsData {
    uk = new GoogleAnalyticsAccount()
    international = new GoogleAnalyticsAccount()
}
