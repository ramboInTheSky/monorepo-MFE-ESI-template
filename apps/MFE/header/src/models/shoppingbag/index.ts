export type PaymentType = "GiftCard" | "Evoucher" | "Card"
export type DiscountsType = "Percentage" | "Cash"
export type PromotionType = "None"

export class BagGetApiModel {
    skipRebuild = false
    ap = false
}

export interface ChargesAndIncentive {
    AdditionalAmount: number
    AdditionalAmountFormatted: string
    AdditionalCode: string
    MaximumOrderValue: number
    MinimumOrderValue: number
    OfferShortDescription: string
    OfferTypeDescription: string
    Type: string
}

export interface MultibuyDiscountedItem {
    DiscountedPrice: number
    MFItemIndex: number
    PromotionCode: string
    QuantityIndex: number
    DiscountedPromotionType: string
    PersonalisedPromotion: boolean
}

export interface FieldValue {
    Field: string
    Value: string
}

export interface Item {
    ItemID: number
    ItemNumber: string
    OptionNo: string
    SizeDescription: string
    Price: string
    Quantity: number
    StockStatus: string
    StockMessage: string
    WarrantyItem: Item[] | null
    Description: string
    Url: string
    PublicationCode: string
    IsSalePrice: boolean
    ProductName: string
    TPSearchDescription: string
    IsMultibuyItem: boolean
    SpecformMessage: string
    CanOrder: boolean
    CanUpdateSize: boolean
    CanUpdateQty: boolean
    MaxQuantity: number
    MultibuyDiscountAppliedItems: MultibuyDiscountedItem[] | null
    ItemCategory: string
    AlternativeDescription: string
    LinkedItems: any[] | null
    GroupID: string | null
    Chain: string | null
    PriceFormatted: string
    TotalPriceFormatted: string
    CanSaveForLater: boolean
    IsDiscount: boolean
    Colour: string | null
    Personalisation: string
    DDFulfiller: string
    PersonalisationReference: string
    PersonalisationFields: FieldValue[] | []
    Fit: string | null
    Department: string
    PersonalisedGiftTheme: string
    CistStoreId: string | null
    IsCistItem: boolean
    CistDate: Date | null
    CistStoreName: string | null
    CistStoreDisplayAddress: string | null
    CistStorePoscode: string | null
    CistBranchNumberPrefix: string | null
    EnglishDescription: string
    EnglishSizeDescription: string
    SaleClearanceSofaImage: string | null
    CistNextOpeningTimes: Record<string, Date>[] | null
    SearchDescription: string | null
    CustomItemReference: string | null
    CustomItemFields: FieldValue[] | []
    FulfilmentType: string | null
    CustomItemType: string | null
    PremiumSiteIndicator: string
    IsHighRiskItem: boolean
    ItemBrand: string | null
    itemImageUrl?: string
}

export interface PersonalisedPromotions {
    Count: number
    PersonalPromotions: string[]
}

export interface Guest {
    IsGuestCheckout: boolean
    HasGuestCheckoutProcessStarted: boolean
    DefaultShippingCharge: string
    TotalCost: string
    ProviderValidation: null
    BagChecksum: number
    UserAgent: string
    ChosenProvider: number
    CanConvert: number
}

export interface Payment {
    Amount: number
    PaymentType: PaymentType
    Type: PaymentType
    LastFourDigits: string
}

export interface Group {
    GroupId: string
    GroupType: string
    PageUrl: string
    DeliveryWeeks: number
}

export interface VipPromotion {
    Code: string
    DiscountValue: number
    DiscountsType: DiscountsType
    ItemPremiumIndicator: string
    PromotionType: PromotionType
    PromotionDescription: string
}

export interface ShoppingBag {
    Anonymous: boolean
    Name: string
    AccountType: string
    Version: number
    ItemCount: number
    Authenticated: boolean
    FinalOrderValue: number
    OrderGoodsValue: number
    Items: Item[]
    ChargesAndIncentives: ChargesAndIncentive[]
    MultiBuyDiscount: number
    MultiBuyDescription: string
    DeliveryCharge: number
    DeliveryDescription: string
    IsShoppingBagFull: boolean
    LowerLimitReached: boolean
    IsPreviewCustomer: boolean
    MultibuyDiscountedPromotionTypes: string
    FromLabelWebsite: boolean
    RunningTotal: number
    Payments: Payment[]
    ItemStateVersion: number
    SessionTransferVariables: string
    AccountNumber: string
    RecreatedFromPreviousSession: boolean
    InvalidItemCount: number
    UserNotifiedOfSessionRecreation: boolean
    UserNotifiedOfInvalidItems: boolean
    OldShoppingBagAge: number
    IsAccountLockedDown: boolean
    Groups: Group[]
    IsCallCentreMode: boolean
    AccountSecurityCheckFailed: boolean
    ForceCashPayment: boolean
    OrderGoodsValueFormatted: string
    DeliveryChargeFormatted: string
    FinalOrderValueFormatted: string
    MultiBuyDiscountFormatted: string
    NotifyEmailExists: boolean
    PaypalSessionAttempts: number
    FirstName: string
    NextUnlimitedStatus: number
    ItemAddedFromSaveForLaterNotification: boolean
    CountryCode: string
    CreditAccountType: number
    MainframeSessionID: string
    ShoppingBagRecentlyRebuilt: boolean
    Xhos: string
    PersonalisedPromotions: PersonalisedPromotions
    Guest: Guest
    VipPromotions: VipPromotion[]
    Children: string[]
    ChildrenCount: number
    ID: string
    Description: string
    Url: string
    Image: string
}
