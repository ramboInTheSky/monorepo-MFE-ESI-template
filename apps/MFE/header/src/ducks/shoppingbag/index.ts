// TODO scaffold shopping bag to be implement in minibag PBI
import {Reducer} from "redux"
import {GetBagCallbackContractModel} from "@monorepo/eventservice"
import {ShoppingBag} from "../../models/shoppingbag"
import type State from "../state"
import {getItemImageUrl} from "../../utils/getMiniShoppingbagItemImageUrl"
import {getOrigin} from "../../utils/getOrigin"
import {excludeIncentives} from "../../utils/excludeIncentives"

export type ShoppingBagDuckState = {
    itemCount: number
    bag: ShoppingBag
    loading: boolean
}

export const SET_BAG = "SET_BAG"

type Action = {
    type: typeof SET_BAG
    payload: ShoppingBagDuckState
}

export const setBag = (payload: ShoppingBagDuckState) => ({
    type: SET_BAG,
    payload,
})

export const updateBag = (getBagData: GetBagCallbackContractModel) => (dispatch: any, getState: any) => {
    const {
        request: {siteUrl},
    } = getState() as State

    if (getBagData.success) {
        const bag = getBagData.data.ShoppingBag
        bag.Items = bag.Items.reverse().map(item => ({
            ...item,
            itemImageUrl: getItemImageUrl(item),
            Url: `${siteUrl ? getOrigin(siteUrl) : ""}${item.Url}`,
        }))
        const itemCount = getBagData.data.ShoppingBag.ItemCount
        if (bag.ChargesAndIncentives && bag.ChargesAndIncentives.length > 0) {
            bag.ChargesAndIncentives = excludeIncentives(bag.ChargesAndIncentives)
        }
        dispatch(setBag({itemCount, bag, loading: false}))
    } else {
        dispatch(setBag(initialState))
    }
}

export const addEVoucherToBag = (getBagData: GetBagCallbackContractModel) => (dispatch: any, getState: any) => {
    const {
        shoppingBag: {bag},
    } = getState() as State
    if (getBagData.success) {
        const bagForEVoucher = getBagData.data.Bag != null ? getBagData.data.Bag : bag
        const {itemCount, items} = bagForEVoucher
        const updatedbag = {items, itemCount, ...bag} as ShoppingBag
        dispatch(setBag({itemCount, bag: updatedbag, loading: false}))
    } else {
        dispatch(setBag(initialState))
    }
}

const initialState: ShoppingBagDuckState = {
    itemCount: 0,
    bag: {} as ShoppingBag,
    loading: true,
}

const reducer: Reducer<ShoppingBagDuckState, Action> = (state = initialState, action) => {
    switch (action.type) {
        case SET_BAG:
            return {...state, ...action.payload}
        default:
            return state
    }
}

export default reducer
