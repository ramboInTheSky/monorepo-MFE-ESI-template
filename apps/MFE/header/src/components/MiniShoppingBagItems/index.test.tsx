import React from "react"
import {render, screen} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {Provider} from "react-redux"
import mockStore, {mockTheme, mockText} from "../../../__mocks__/mockStore"
import {Item, ShoppingBag} from "../../models/shoppingbag"
import {MiniShoppingBagItems, MiniShoppingBagItemsProps} from "."

describe("Components - MiniShoppingBagItems: ", () => {
    describe("Items in the Bag- MiniShoppingBagItems: ", () => {
        let props: MiniShoppingBagItemsProps
        beforeEach(() => {
            props = {
                bag: {
                    Description: "Excluding UK Standard Delivery (Normally £3.99)",
                    MultiBuyDiscount: 1,
                    FinalOrderValueFormatted: "£89.00",
                    MultiBuyDiscountFormatted: "£1.00",
                    Items: [
                        {
                            ItemID: 1,
                            ItemNumber: "436441",
                            OptionNo: "04",
                            SizeDescription: "10 S",
                            StockMessage: "In Stock",
                            Description: "UGO Herman Melange Jacket",
                            AlternativeDescription: "",
                            Price: "45.00",
                            Quantity: 5,
                            CistStoreName: "LEICESTER - HIGH CROSS SHOPPING CENTRE",
                        } as Item,
                        {
                            ItemID: 2,
                            ItemNumber: "436461",
                            OptionNo: "04",
                            SizeDescription: "10 S",
                            StockMessage: "Delayed",
                            Price: "45.00",
                            Description: "UGO Herman Melange Jacket",
                            AlternativeDescription: "",
                            Quantity: 5,
                            CistStoreName: "",
                        } as Item,
                    ],
                } as ShoppingBag,
                text: mockText.miniShoppingBag,
            }
        })

        it("should match the snapshot ", () => {
            const {asFragment} = render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...props} />
                    </SCThemeProvider>
                </Provider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })

        it("should display the multibuy discount", () => {
            render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...props} />
                    </SCThemeProvider>
                </Provider>,
            )
            const discountExpected = screen.getByTestId("header-mini-shopping-bag-multibuy-discount").textContent
            expect(discountExpected).toBe(`-${props.bag.MultiBuyDiscountFormatted}`)
        })

        it("should not display the multibuy discount", () => {
            const innerProps = {
                bag: {
                    Description: "Excluding UK Standard Delivery (Normally £3.99)",
                    MultiBuyDiscount: 0,
                    FinalOrderValueFormatted: "£45.00",
                    MultiBuyDiscountFormatted: "",
                    Items: [
                        {
                            ItemID: 1,
                            ItemNumber: "436441",
                            OptionNo: "04",
                            SizeDescription: "10 S",
                            StockMessage: "In Stock",
                            Description: "UGO Herman Melange Jacket",
                            AlternativeDescription: "",
                            Price: "45.00",
                            Quantity: 5,
                            CistStoreName: "LEICESTER - HIGH CROSS SHOPPING CENTRE",
                        } as Item,
                    ],
                } as ShoppingBag,
                text: mockText.miniShoppingBag,
                template: "default",
            }
            render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...innerProps} />
                    </SCThemeProvider>
                </Provider>,
            )
            const totalExpected = screen.getByTestId("header-mini-shopping-bag-total").textContent
            expect(totalExpected).toBe(`${innerProps.bag.FinalOrderValueFormatted}`)

            const discountElement = screen.queryAllByTestId("header-mini-shopping-bag-multibuy-discount")
            expect(discountElement.length).toBe(0)
        })

        it("should display the total", () => {
            render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...props} />
                    </SCThemeProvider>
                </Provider>,
            )
            const totalExpected = screen.getByTestId("header-mini-shopping-bag-total").textContent
            expect(totalExpected).toBe(`${props.bag.FinalOrderValueFormatted}`)
        })

        it("should add event listener wheel to bagContainerRef", () => {
            // eslint-disable-next-line @typescript-eslint/no-empty-function
            const useEffect = jest.spyOn(React, "useEffect").mockImplementation(() => {})
            render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...props} />
                    </SCThemeProvider>
                </Provider>,
            )
            expect(useEffect).toHaveBeenCalled()
        })

        it("should remove event listener wheel from bagContainerRef", () => {
            // eslint-disable-next-line @typescript-eslint/no-empty-function
            const useEffect = jest.spyOn(React, "useEffect").mockImplementation(() => {})
            const component = render(
                <Provider store={mockStore}>
                    <SCThemeProvider theme={mockTheme}>
                        <MiniShoppingBagItems {...props} />
                    </SCThemeProvider>
                </Provider>,
            )

            component.unmount()
            expect(useEffect).toHaveBeenCalled()
        })
    })
})
