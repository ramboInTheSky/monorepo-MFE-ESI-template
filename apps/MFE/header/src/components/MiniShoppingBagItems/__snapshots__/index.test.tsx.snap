// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - MiniShoppingBagItems:  Items in the Bag- MiniShoppingBagItems:  should match the snapshot  1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-top: 0.0625rem solid #eaeaea;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: 0.0075rem;
  -moz-letter-spacing: 0.0075rem;
  -ms-letter-spacing: 0.0075rem;
  letter-spacing: 0.0075rem;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  height: 2.75rem;
  width: 5.375rem;
}

.c4 {
  padding: 1rem;
}

.c8 {
  white-space: nowrap;
  padding-right: 0.938rem;
}

.c9 {
  font-family: "AzoSansRegular","Helvetica Neue",Arial,sans-serif;
  margin: 0.25rem 0 0 0;
  font-size: 0.875rem;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  color: #d88f31;
}

.c6 {
  padding: 1rem 0;
  width: 100%;
}

.c6.c6 > p {
  font-family: "AzoSansRegular","Helvetica Neue",Arial,sans-serif;
  color: #515151;
  word-break: break-all;
  padding-right: 0.613rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 > h3:first-child {
  word-wrap: break-word;
  max-width: 10.3125rem;
  padding-right: 0.625rem;
}

.c7 > h3:last-child {
  white-space: nowrap;
}

.c0 {
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0.25rem;
  background-color: #f7f7f7;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  font-family: "AzoSansMedium","Helvetica Neue",Arial,sans-serif;
  font-size: 0.875rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem 0 1rem;
}

.c12 {
  font-size: 0.75rem;
  padding: 0 1rem 0.5rem 1.125rem;
  border-bottom: .0625rem solid #9e9e9e;
}

.c12.c12 > p {
  font-size: 0.75rem;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.5;
  -webkit-letter-spacing: 0.0075rem;
  -moz-letter-spacing: 0.0075rem;
  -ms-letter-spacing: 0.0075rem;
  letter-spacing: 0.0075rem;
  word-break: break-all;
}

.c1 {
  min-width: 100%;
  overflow-y: auto;
  -ms-overflow-style: auto;
  -webkit-scrollbar-width: auto;
  -moz-scrollbar-width: auto;
  -ms-scrollbar-width: auto;
  scrollbar-width: auto;
  max-width: 19.875rem;
  border-bottom: 0.0625rem solid #eaeaea;
  background-color: #f7f7f7;
}

.c1 > div:first-child {
  border-top: 0;
}

.c1::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 0.475rem;
}

.c1::-webkit-scrollbar-thumb {
  border-radius: 0.25rem;
  background-color: rgba(0,0,0,0.5);
  box-shadow: 0 0 1px rgba(255,255,255,0.5);
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="header-mini-shopping-bag-items"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="undefined#436441"
        >
          <div
            class="c4"
          >
            <img
              alt="UGO Herman Melange Jacket"
              class="c5"
              data-testid="header-mini-shopping-bag-item-image"
            />
          </div>
        </a>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
              data-testid="header-mini-shopping-bag-item-description"
            >
              UGO Herman Melange Jacket
            </h5>
            <div
              class="c8"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignRight"
              />
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Size: 10 S
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Qty: 5
          </p>
          <div
            class="c9"
          >
            In Stock at LEICESTER - HIGH CROSS SHOPPING CENTRE
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <a
          class="c3"
          href="undefined#436461"
        >
          <div
            class="c4"
          >
            <img
              alt="UGO Herman Melange Jacket"
              class="c5"
              data-testid="header-mini-shopping-bag-item-image"
            />
          </div>
        </a>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
              data-testid="header-mini-shopping-bag-item-description"
            >
              UGO Herman Melange Jacket
            </h5>
            <div
              class="c8"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignRight"
              />
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Size: 10 S
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Qty: 5
          </p>
          <div
            class="c9"
          >
            Delayed 
          </div>
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Promotional Savings
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5"
        data-testid="header-mini-shopping-bag-multibuy-discount"
      >
        -£1.00
      </h5>
    </div>
    <div
      class="c11"
    >
      <span>
        Total
      </span>
      <span
        data-testid="header-mini-shopping-bag-total"
      >
        £89.00
      </span>
    </div>
    <div
      class="c12"
    >
      <span>
        Excluding UK Standard Delivery (Normally £3.99)
      </span>
    </div>
  </div>
</DocumentFragment>
`;
